context-propagation/src/main/java/io/micrometer/context/ContextExecutorService.java

```diff
@@ -34,19 +34,19 @@
  * @author Rossen Stoyanchev
  * @since 1.0.0
  */
-final class ContextExecutorService implements ExecutorService {
+class ContextExecutorService<EXECUTOR extends ExecutorService> implements ExecutorService {
```

Comment:
Maybe we can also rename this to `ContextPropagatingExecutorService` and likewise for the subclass? The `Context` prefix has been catching me out, it's a little too unspecific.